### ISSUE TITLE
`UserDefaults.Dependency.previewValue` leaks to other `withDependencies`'s?

### ISSUE THREAD

**Issue by Zeta611** [tgrapperon/swift-dependencies-additions#33]
Hello, I've found out that `UserDefaults.Dependency.previewValue` does not re-initialize and leaks to other `withDependencies`'s.

Here is a MWE. With the following view / model pair,
```swift
final class ContentViewModel: ObservableObject {
    @Published var text1 = "nil"
    @Published var text2 = "nil"
    @Dependency.AppStorage("myDefault1") private var myDefault1: String?
    @Dependency.AppStorage("myDefault2") private var myDefault2: String?
    init() {
        if let myDefault1 { text1 = myDefault1 }
        if let myDefault2 { text2 = myDefault2 }
    }
}
struct ContentView: View {
    @ObservedObject private(set) var viewModel: ContentViewModel
    var body: some View {
        VStack {
            Text(viewModel.text1)
            Text(viewModel.text2)
        }
    }
}
```
when I setup a preview as:
```swift
struct ContentView_Previews: PreviewProvider {
    static var previews: some View {
        ContentView(viewModel: withDependencies {
//            $0.userDefaults = .previewValue
            $0.userDefaults.set("v1", forKey: "myDefault1")
        } operation: {
            ContentViewModel()
        })
        ContentView(viewModel: withDependencies {
//            $0.userDefaults = .previewValue
            $0.userDefaults.set("v2", forKey: "myDefault2")
        } operation: {
            ContentViewModel()
        })
    }
}
```
the two `ContentView`s display the same content, which means that the `.previewValue` is not being reset. Removing either of the comments work as (I have) expected.

I'm not 100% sure if this is a bug, but it seems misleading because the documentation for `withDependencies` says (emphasis mine):
> `updateValuesForOperation` A closure for updating the current dependency values for the **duration** of the operation.

I initially discovered this behavior testing a somewhat complex logic---involving asynchronous codes manipulating user default key-value pairs---, and it took quite a bit of time to locate the cause.

Should this be considered a bug, or an expected behavior?
Thanks!

**Comment by tgrapperon** [comment]
Hey @Zeta611! This is unfortunate indeed. Sorry about the bug you had. In this fragment of code:
```swift
  ContentView(viewModel: withDependencies {
//  $0.userDefaults = .previewValue
    $0.userDefaults.set("v1", forKey: "myDefault1")
} operation: {
  ContentViewModel()
})
```
You're not updating the dependency itself, you're simply calling one of its endpoints.

As a default behavior of `Dependencies`, default values are captured on first access, and the `.ephemeral` `previewValue` is shared among previews if you're not effectively overriding it, even if the endpoints you're calling are giving you the impression that you are changing things. And the fact that previews are continuously (but partially) re-created doesn't help.

But this is maybe something that can be improved for previews, as each one should indeed be considered independent. We can maybe effectively reset the cache (or not use it) in preview contexts. I'll think about it a little more.

So, in other words, it's behaving as expected, but the behavior can maybe be tweaked in preview contexts. As an alternative to this specific problem, a version of `.ephemeral` with a closure to configure its initial values would probably nudge you toward creating a new `userDefaults` instance.

**Comment by Zeta611** [comment]
Thank you for the confirmation, and for always providing a detailed guide.
I'll stick with reinitializing the dependencies when it's required for now :)
